epoxy-processor/src/main/java/com/airbnb/epoxy/ClassToGenerateInfo.java

```diff
@@ -50,7 +53,22 @@ public ClassToGenerateInfo(TypeElement originalClassName, ClassName generatedCla
           && !subElement.getModifiers().contains(Modifier.PRIVATE)) {
         List<? extends TypeMirror> params = subElement.asType().accept(CONSTRUCTOR_VISITOR, null);
         constructors
-            .add(new ConstructorInfo(subElement.getModifiers(), buildConstructorParamList(params)));
+            .add(new ConstructorInfo(subElement.getModifiers(), buildParamList(params)));
+      }
+    }
+
+    // Get information about methods returning class type of the original class so we can
+    // duplicate them in the generated class for chaining purposes
+    for (Element subElement : originalClassName.getEnclosedElements()) {
+      if (subElement.getKind() == ElementKind.METHOD
+          && ((ExecutableType) subElement.asType()).getReturnType()
+            .equals(originalClassName.asType())
+          && !subElement.getModifiers().contains(Modifier.PRIVATE)
+          && !subElement.getModifiers().contains(Modifier.STATIC)) {
+        List<? extends TypeMirror> params = ((ExecutableType) subElement.asType())
+            .getParameterTypes();
+        methodsReturningClassType.add(new MethodInfo(subElement.getSimpleName().toString(),
+            subElement.getModifiers(), buildParamList(params)));
```

Comment:
could pull out `subElement.getModifiers()` to a variable since it is called a few times